src/components/List/List.tsx

```diff
@@ -1,90 +1,157 @@
-      dataTestId,
-      ...rest
-    },
-    ref
-  ) => {
-    return (
-      <div css={wrapperStyle({ width, isSearchable })}>
-        {isVirtualized ? (
-          <VirtualizedList
-            items={data}
-            rowSize={rowSize}
-            customWidth={width}
-            customHeight={height}
-            ref={ref}
-            selectedItem={selectedItem}
-            defaultOption={defaultOption}
-            searchTerm={searchTerm}
-            handleOptionClick={handleOptionClick}
-            dataTestId={dataTestId}
-            {...rest}
-          />
-        ) : (
-          <NormalList
-            items={data}
-            rowSize={rowSize}
-            width={width}
-            height={height}
-            ref={ref}
-            selectedItem={selectedItem}
-            isSearchable={isSearchable}
-            defaultOption={defaultOption}
-            searchTerm={searchTerm}
-            handleOptionClick={handleOptionClick}
-            dataTestId={dataTestId}
-            {...rest}
-          />
-        )}
+const List = React.forwardRef<HTMLUListElement, ListProps>((props, ref) => {
+  const {
+    width,
+    height,
+    isVirtualized = false,
+    // isSearchable,
+    // searchTerm,
+    dataTestId,
+  } = props;
+  /** ignore the change on the `onSelectionChange` from any to unknown **/
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const state = useListState<SelectOption>({ ...props, selectionMode: 'single' });
+  const myRef = React.useRef(null);
+  const combinedRefs = useCombinedRefs(myRef, ref);
+  /** ignore the change on the `onSelectionChange` from any to unknown **/
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const { listBoxProps } = useListBox(props, state, combinedRefs);
+
+  const firstKey = state.collection.getFirstKey();
+  const first = firstKey ? state.collection.getItem(firstKey) : null;
+
+  return (
+    <div css={wrapperStyle({ width })}>
+      <div data-testid={dataTestId ? `${dataTestId}_list` : 'ictinus_list'}>
+        <div style={{ height: height || '100%' }}>
+          <Window
+            {...listBoxProps}
+            css={listStyle({ width, height })}
+            id={listBoxProps.id}
+            isVirtualizationEnabled={isVirtualized}
+            rowHeight={first?.props.rowSize === 'compact' ? 40 : 56}
+            ref={combinedRefs}
+          >
+            {Array.from(state.collection).map((item) => {
+              return item.type === 'section' ? (
+                <ListBoxSection key={item.key} section={item} state={state} />
+              ) : (
+                <Option key={item.key} item={item} state={state} />
+              );
+            })}
+          </Window>
+        </div>
       </div>
-    );
-  }
-);
+    </div>
+  );
+});
 List.displayName = 'List';
 
-export default memo(List, isEqual);
+function Option({ item, state, style }: { item: any; state: any; style?: any }) {
```

Comment:
this could be moved to a `subcomponents` file

Reply:
The whole `List` component is an overlay layer for the Aria. The `Option` is to use `ListItemWrapper`, `ListBoxSection` is to user the group section.  I feel that split them to separate files will add unnecessary complexity when reading. I don't see how splitting could help us here

Adding more documentation here at [`3b16bc4` (#655)](https://github.com/Orfium/orfium-ictinus/pull/655/commits/37ecdeae6f568ec5c3495ef76e0914fa9062e595)

Reply:
ok I see your point, fair enough